src/tests/ftest/control/dmg_nvme_scan_test.py

```diff
@@ -0,0 +1,95 @@
+     http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+kworker/16:1
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+import time
+
+from dmg_utils import DmgCommand
+from server_utils import ServerCommand
+from apricot import TestWithoutServers
+from avocado.utils import process
+
+class DmgNvmeScanTest(TestWithoutServers):
+    """ Simple test to verify the scan function of the dmg tool.
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        super(DmgNvmeScanTest, self).__init__(*args, **kwargs)
+
+    def clean_up(self):
+        """Setup/cleanup for the daos_server to run properly."""
+        cmd_list = [
+            "umount /mnt/daos; rm -rf /mnt/daos",
+            "rm -rf /tmp/daos_sockets/",
+            "rm -rf /tmp/*.log",
+        ]
+        for cmd in cmd_list:
+            try:
+                process.run(cmd, verbose=True, sudo=True)
+            except process.CmdError as error:
+                self.fail("Error issuing '{}': {}".format(cmd, error))
+
+    def test_dmg_nvme_scan_basic(self):
+        """ Test basic dmg functionality to scan nvme the storage on system.
+        :avocado: tags=all,hw,dmg,control
+        """
+        self.clean_up()
+
+        # Create daos_server command
+        server = ServerCommand()
+        server.get_params(self, "/run/daos_server/*")
+
+        # Edit the config value to full path
+        server.config.value = os.path.join(
+            self.basepath, str(server.config.value))
+
+        # Run server as a background process.
+        svr_subproc = server.subproc()
+        try:
+            pid = svr_subproc.start()
+        except process.CmdError as details:
+            self.fail("Server command failed: {}".format(details))
+
+        # Wait for server to start
+        time.sleep(3)
+
+        # Create daos_shell command
+        dmg = DmgCommand("daos_shell")
+        dmg.get_params(self, "/run/daos_shell/*")
+
+        # Get server for daos_shell command and assign hostlist value
+        test_machines = self.params.get("test_machines", "/run/hosts/*")
+        for i, machine in enumerate(test_machines):
+            test_machines[i] = "{}:{}".format(machine, dmg.port.value)
+
+        dmg.hostlist.value = ",".join(test_machines)
```

Comment:
Obtain the port number directly from the test yaml and remove the port attribute from DmgCommand:
```
        port = self.params.get("port", "/run/hosts/*")
        dmg.hostlist.update(",".join(["{}:{}".format(host, port) for host in test_machines])
```

Reply:
Will change.